barretenberg/cpp/src/barretenberg/vm2/constraining/testing/check_relation.hpp

```diff
@@ -27,13 +46,26 @@ void check_relation_internal(const Trace& trace, std::span<size_t> subrelations)
             }
         }
     }
+    // For all subrelations, the result should be zero at the end of the trace.
```

Comment:
I changed this to work both with linearly dependent or independent subrelations. The dependent ones are only checked at the end.